### PR TITLE
Server database

### PR DESCRIPTION
**Pull request by ZQUEMA**
Implement functions to easily use the database

### CODE REVIEW

File: libs/database/search/channel_search.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
+** EPITECH PROJECT, 2022
+** Teams
+** File description:
+** channel_searchs
+*/
+
+#include "../fields/channel.h"
```
Comment (TomDUVAL-MAHE) [inline libs/database/search/channel_search.c:8]: Path en dure !

**Review by TomDUVAL-MAHE — COMMENTED** [review]
LES VARIABLES !!!

File: libs/database/search/discussion_search.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#include "../fields/discussion.h"
+#include <stdbool.h>
+#include <string.h>
+
+bool discussion_id_compare(const void *elem, const void *search)
+{
+    const discussion_t *u = (const discussion_t *)elem;
+    const id_t *id = (const id_t *)search;
+
+    if (u->id == *id)
```
Comment (aurelenc) [inline libs/database/search/discussion_search.c:17]: return (u->id == *id);

File: libs/database/search/discussion_search.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        return true;
+    return false;
+}
+
+bool discussion_name_compare(const void *elem, const void *search)
+{
+    const discussion_t *u = (const discussion_t *)elem;
+    const char *name = (const char *)search;
+
+    if (strcmp(u->name, name) == 0)
```
Comment (aurelenc) [inline libs/database/search/discussion_search.c:27]: return (!strcmp(u->name, name));

File: libs/database/search/discussion_search.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        return true;
+    return false;
+}
+
+bool discussion_user_one_id_compare(const void *elem, const void *search)
+{
+    const discussion_t *u = (const discussion_t *)elem;
+    const id_t *id = (const id_t *)search;
+
+    if (u->user_one_id == *id)
```
Comment (aurelenc) [inline libs/database/search/discussion_search.c:37]: return (u->user_one_id == *id);

**Review by aurelenc — APPROVED** [review]

**Pull request merged**